src/server/scripts/Spells/spell_paladin.cpp

```diff
@@ -357,6 +358,38 @@ class spell_pal_blinding_light : public SpellScript
     }
 };
 
+// 1022 - Blessing of Protection
+class spell_pal_blessing_of_protection : public SpellScript
+{
+    PrepareSpellScript(spell_pal_blessing_of_protection);
+    
+    bool Validate(SpellInfo const* /*SpellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_PALADIN_FORBEARANCE });
+    }
+    SpellCastResult Forbearance()
+    {
+        if (Unit* caster = GetCaster())
+        {
+            if(caster->HasAura(SPELL_PALADIN_FORBEARANCE))
+                return SPELL_FAILED_TARGET_AURASTATE;
+        }
+        return SPELL_CAST_OK;
+    }
+    
+    void HandleOnHit()
+    {
+        if (Unit* caster = GetCaster())
```

Comment:
SpellScript, Caster is never null =)

Reply:
if its null so how it work inside game?

Reply:
spell caster cannot be null, because that would mean the spell does not get cast in the first place.

Reply:
Script is called only if has caster...
so you dont need a null check for caster in SpellScripts.
But it can be null in AuraScripts xD

Reply:
if (getcaster()) so if there is no caster it wont run or its not?